app/Platform/src/Http/Controllers/RegisterMemberController.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace App\Platform\Http\Controllers;
+
+use App\Framework\Enums\RoleName;
+use App\Framework\Models\User;
+use App\Platform\Http\Requests\RegisterRequest;
+use App\Platform\Http\Resources\MemberProfileResource;
+use Illuminate\Auth\Events\Registered;
+use Illuminate\Support\Facades\Hash;
+use Knuckles\Scribe\Attributes\BodyParam;
+use Knuckles\Scribe\Attributes\Endpoint;
+use Knuckles\Scribe\Attributes\Group;
+
+class RegisterMemberController
+{
+    #[Group('Platform')]
+    #[Endpoint(title: 'Sign Up', description: 'This endpoint is for registering new members')]
+    #[BodyParam('email', 'string', required: true, example: "example@test.com")]
+    #[BodyParam('password', 'string', required: true, example: "password")]
+    #[BodyParam('first_name', 'string', required: true, example: "John")]
+    #[BodyParam('last_name', 'string', required: true, example: "Doe")]
+    #[BodyParam('image', 'file', required: false)]
+    public function store(RegisterRequest $request)
+    {
+        $user = User::query()->create([
```

Comment:
Wrap this code within a database transaction so that if it fails, we don't want to end up with a broken state in the database.

Also, you can move this code within a service class to simplify the controller.